torch/backends/cudnn/rnn.py

```diff
@@ -190,7 +196,11 @@ def forward(fn, input, hx, weight, output, hy):
         handle = cudnn.get_handle()
         fn.datatype = cudnn._typemap[input.type()]
         is_input_packed = fn.batch_sizes is not None
-
+        if is_input_packed:
+            warnings.warn(
+                "Warning: persistent algorithm not supported for variable length input."
+                "Switching to standard")
+            fn.persistent = False  # persistent algo is not supported for variable length input
```

Comment:
I think both defaulting to non-persistent and picking persistent kernels for batch sizes where we're pretty sure that it's going to be faster are both ok. Do these kernels really have such unpredictable perf?

We don't have any function for querying the cc.

I don't think that the overhead is going to be significant, as far as I know most commonly used sequences are no longer than a few thousand steps, so if we were to use the fixed bin size of 50, we'd have around 100 evaluations. That seems reasonable to me, but I agree that 50 is quite arbitrary and it would be better to design it in a way that better fits the perf characteristics.

An example would be to put all sequence lengths for which we're 99% sure that it's slower in a single bin or even exclude that form benchmarks, use smaller bins on the boundary where persistent kernels might be advantageous, and gradually enlarge the bin size once we enter the region where the sequences are very long and we're pretty sure that persistent RNNs will help.